deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/LayerConfiguration.java

```diff
@@ -0,0 +1,313 @@
+package org.deeplearning4j.nn.modelimport.keras;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.deeplearning4j.nn.conf.layers.*;
+import org.deeplearning4j.nn.weights.WeightInit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Routines for importing saved Keras layer configurations.
+ * @author davekale
+ */
+public class LayerConfiguration {
+    private static Logger log = LoggerFactory.getLogger(LayerConfiguration.class);
+
+    private LayerConfiguration() {}
+
+    /**
+     * Configure DL4J Layer from a Keras layer configuration.
+     *
+     * @param kerasLayerClass  String containing the Keras layer class type
+     * @param kerasConfig      Map containing Keras layer properties
+     * @return                 DL4J Layer configuration
+     * @see Layer
+     */
+    public static Layer buildLayer(String kerasLayerClass, Map<String,Object> kerasConfig) {
+        return buildLayer(kerasLayerClass, kerasConfig, false);
+    }
+
+    /**
+     * Configure DL4J Layer from a Keras layer configuration.
+     *
+     * @param kerasLayerClass  String containing the Keras layer class type
+     * @param kerasConfig      Map containing Keras layer properties
+     * @param isOutput         Whether this is an output layer
+     * @return                 DL4J Layer configuration
+     * @see Layer
+     */
+    public static Layer buildLayer(String kerasLayerClass, Map<String,Object> kerasConfig, boolean isOutput) {
+        Layer layer = null;
+        switch (kerasLayerClass) {
+            case "Dense":
+            case "TimeDistributedDense":
+                layer = buildDenseLayer(kerasConfig);
+                break;
+            case "LSTM":
+                layer = buildGravesLstmLayer(kerasConfig);
+                break;
+            case "Convolution2D":
+                layer = buildConvolutionLayer(kerasConfig);
+                break;
+            case "MaxPooling2D":
+                layer = buildSubsamplingLayer(kerasConfig);
+                break;
+            case "Flatten":
+                log.warn("DL4J adds reshaping layers during model compilation");
+                break;
```

Comment:
@agibsonccc Adding a link to github repos location showing code where we add preprocessing layers: https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/MultiLayerConfiguration.java#L429

Does that suffice?